src/components/common/CodeReviewTools.js

```diff
@@ -0,0 +1,401 @@
+    };
+  })();
+
+  handleEdit = () => {
+    const { dispatchOpenEditorView } = this.props;
+    // change layout to edit mode
+    dispatchOpenEditorView();
+  };
+
+  handleHiddenToggle = (visibility) => {
+    const { hideCommentsCallback } = this.props;
+    // send true if the visibility is not "visible" -> the comments are hidden
+    hideCommentsCallback(visibility !== 'show');
+  };
+
+  handleChangeBranch = (value) => {
+    const { codeVersions, codeContributors } = this.props;
+    const { selectedVersion } = this.state;
+    const availableCodeVersions = codeVersions[codeContributors.indexOf(value)];
+    // check if the new set of choices includes the previously selected code version
+    if (!availableCodeVersions.includes(selectedVersion)) {
+      // instructor is selected and there is only one version available
+      if (value.value === DEFAULT_CODE_ID) {
+        this.handleChangeVersion(availableCodeVersions[0]);
+      } else {
+        this.setState({ selectedVersion: null });
+      }
+    }
+    // change the code that is displayed and filter the history select
+    this.setState({ selectedBranch: value, availableCodeVersions });
+  };
+
+  handleChangeVersion = (value) => {
+    const { dispatchSetCodeEditorSettings } = this.props;
+    // change the code that is displayed and filter the history select
+    this.setState({ selectedVersion: value });
+    // this.setState({ selectedBranch: value })
+    dispatchSetCodeEditorSettings(value.value);
+  };
+
+  handleInfo = () => {
+    const { dispatchOpenCommitInfoDialog } = this.props;
+    dispatchOpenCommitInfoDialog();
+  };
+
+  renderTopBar = () => {
+    const {
+      t,
+      classes,
+      allVisibleState,
+      showVisibilityButton,
+      showEditButton,
+      codeContributors,
+      showHistoryDropdown,
+    } = this.props;
+    const { selectedBranch, selectedVersion, availableCodeVersions } =
+      this.state;
+    const selectedVisibility = allVisibleState ? 'hide' : 'show';
+
+    const hideAllCommentsToggleControl = (
+      <ToggleButton
+        className={classes.toggleButton}
+        size="small"
+        value={selectedVisibility}
+        onChange={() => this.handleHiddenToggle(selectedVisibility)}
+      >
+        {allVisibleState ? (
+          <VisibilityOffRounded fontSize="small" />
+        ) : (
+          <VisibilityRounded fontSize="small" />
+        )}
+      </ToggleButton>
+    );
+
+    const editButtonControl = (
+      <ToggleButton
+        className={classes.editButton}
+        variant="outlined"
+        size="small"
+        value="edit"
+        onClick={this.handleEdit}
+      >
+        <EditRounded fontSize="small" />
+      </ToggleButton>
+    );
+
+    const infoButtonControl = (
+      <ToggleButton
+        className={classes.infoButton}
+        variant="outlined"
+        size="small"
+        value="info"
+        onClick={this.handleInfo}
+        // disabled={selectedBranch.value === DEFAULT_CODE_ID}
```

Comment:
Do we still need this?

Reply:
I don't think so as i made a default commit info for the instructor code.